ivy/functional/frontends/jax/devicearray.py

```diff
@@ -1,18 +1,79 @@
+# global
+import functools
+
+
 # local
 import ivy
 import ivy.functional.frontends.jax as jax_frontend
 
 
+def _frontend_wrapper(f):
```

Comment:
this should be handled in a fully recursive manner using something in the `ivy.nest` module. Could you also please add a comment explaining why this function is necessary at all?

Reply:
I think because it's a new class to resemble the native frontend behaviour (not ivy array nor native classes) it doesn't access the data automatically (since we have decorator or similar handling in the backend)

it seems to be trying to operate on the class object itself, therefore throwing various errors if the second onwards argument is the frontend type (first argument doesn't matter as we pass it as `self.data`)

so I've created a decorator to resolve the issue, but on second thought, we might need to add the frontend class type to the backend conversions as well so that they're recognised and handled properly (else, there may be certain class that we could inherit from..?)